### ISSUE TITLE
Collaboration?

### ISSUE THREAD

**Issue by alphapapa** [TannerBaldus/getlucky#3]
Hi, I recently put up https://github.com/alphapapa/pyza, a console-based Songza player.  I wonder if we could share some code (or, at least, if I could use some of yours) for finding and selecting playlists.  What I have just does a station name/description search, but I'm interested in adding some mood-type searching too.  What do you think?  I looked at using some of your code a few weeks ago, but I wasn't quite sure where to begin separating out what I would need.

Thanks for any ideas.

**Comment by TannerBaldus** [comment]
Hey  pyza looks amazing! 
Okay so the way I was getting moods playlists was by parsing the html pages of one of songza's mood pages like [this](http://songza.com/discover/moods/energetic/). I was searching for ```li``` tags of the class ```browse-playlist playable```. However that way wouldn't be particularly useful for your implementation of finding stations of going thru a json of station ids. 

But I was looking through [the html source](http://pastebin.com/GNb9c8Ki) of the Songza pages and found the some a useful json that we can parse from the page (line 225).

I was able to get to that text using  convert it into a dict using demsjon as python's built in json was too strict for the json on line 225.  If you don't want to add that extra dependency I'm sure you could use a regex to fix the json structure.  Namely converting all of the keys that are just ints to surround them with quotes.

I forked your repo and I should have the functions for searching by mood in your Songza class soon.

**Comment by alphapapa** [comment]
Hey, that's fantastic!  I just pulled your commits and merged them with a bit of refactoring and restyling.  I hope I didn't mangle your work too much, I just tried to reduce the size of the diff.  :)  I'm also new to collaborating with git, so I hope my merging etiquette was all right.  =)

Thanks a lot for doing this!  It must have taken a while to dig through the HTML and figure out how to fix up the JSON so it would be usable.  I did run into a JSON parsing error with a query while I was working on it, but I don't remember what triggered it, so maybe I'll run into it again in the future.  But this seems to work great!

If you're interested in doing any more, it would be nice to be able to display a list of activities, genres, and moods, so people don't have to either guess blindly or look them up on the web site.  If you don't want to do this, I'll probably get around to it sometime.  :)

Thanks again, this makes it so much more awesome!

**Comment by TannerBaldus** [comment]
Sweet! Actually it wasn't that bad. After seeing your code it kind of jumped out at me because I knew what I was looking for.


As for your etiquette I think you're fine :). It's ultimately your project and you know what you want it to look like more than anybody.
However I would argue for the smaller split up functions as it would make things easier to add new features and test. That said the combining of the two searches looks good right now but I do think it'd be nicer for adding new search modes with split up methods.

Now what I do feel slightly stronger about is re factoring the argument parsing into the findStations out into a different function. Also I'd add an example call using the short form category to the readme. e.g. 
```pyza -f m:earthy```.


And for the list we could probably go about that one of two ways. 
* We could use BeautifulSoup (Speaking of, you asked about it in the comments and I originally used that but found I could do without adding the extra dependency.) and parse the html to get all of the activities on each time they ask for it.

* Could just do the above once and cache it. I prefer this way as I don't think they change that much and it would be faster.

**Comment by alphapapa** [comment]
Thanks, it was really nice to see you figure out how to integrate your code into mine, and a lot less effort (haha)!

I thought about keeping the search functions split, but since there wasn't anything else that would call that code, I decided to combine it.  But I hadn't thought about testing, and you're right there.  (I'm still a Python novice, and I haven't really gotten into testing yet.)

About findStations, I'll need to look at that again to see what you mean...

For the listing, well, I've almost finished that, so look for a commit soon.  :)

Thanks again.

**Comment by alphapapa** [comment]
For `findStations`, do you mean separating the code that parses the category out of a query string into a separate function, so that you'd call it like `findStations(query, category)`?  If so, I think I understand why, but since no other function needs to do that, I think I'd rather keep it all in `findStations`.  That way all you have to do is pass a simple string to it and it will give you results.  If another function needs to separate the category out of the query in the future, I would be glad to factor that out into a separate function, but I think I'd still prefer to leave `findStations` accepting one argument.

Do you see what I mean?  Are you thinking of testing the category-separating code?  That seems like a good reason, but since it's so simple, I'm not sure it's worth testing...  :)

Thanks.

**Comment by TannerBaldus** [comment]
Yeah, I was just thinking that we'd pull the parses the category out from the ```findStations``` and just use that function in ```findStations``` . So we only pass in one string to ```findStations``` but still have our concerns separated.  So something like:

```
def findStations(query):
    query, category = Songza._parseQuery(query)
    if category:
```
etc.

**Comment by alphapapa** [comment]
That sounds good to me.  Feel free to go ahead and do it, or I'll probably do it eventually.  :)  Thanks.
